plan_delegator/src/plan_delegator.cpp

```diff
@@ -172,6 +187,8 @@ namespace plan_delegator
                 ++current_maneuver_index;
                 continue;
             }
+
+            updateManeuverDistances(maneuver);
```

Comment:
I think we need a check here to delete the maneuver if the new end distance is less than the vehicle's current location after adjustment. In that case it should be removed from the list of the maneuvers to be planned as it's already complete.

Reply:
updated